### ISSUE TITLE
Failed to add rule for system call (many funcs)()

### ISSUE THREAD

**Issue by ghost** [systemd/systemd#8790]
### Submission type

 - Bug report

### systemd 238 the issue has been seen with

Here is a log from dmesg:

```
[   11.931675] systemd[193]: Failed to add rule for system call tuxcall() / 184, ignoring: Numerical argument out of domain
[   11.932012] systemd[193]: Failed to add rule for system call security() / 185, ignoring: Numerical argument out of domain
[   11.932126] systemd[193]: Failed to add rule for system call s390_runtime_instr() / -10196, ignoring: Numerical argument out of domain
[   11.932260] systemd[193]: Failed to add rule for system call rtas() / -10187, ignoring: Numerical argument out of domain
[   11.932603] systemd[193]: Failed to add rule for system call s390_pci_mmio_write() / -10198, ignoring: Numerical argument out of domain
[   11.932696] systemd[193]: Failed to add rule for system call pciconfig_write() / -10088, ignoring: Numerical argument out of domain
[   11.932956] systemd[193]: Failed to add rule for system call pciconfig_read() / -10087, ignoring: Numerical argument out of domain
[   11.933039] systemd[193]: Failed to add rule for system call s390_pci_mmio_read() / -10197, ignoring: Numerical argument out of domain
[   11.933124] systemd[193]: Failed to add rule for system call pciconfig_iobase() / -10086, ignoring: Numerical argument out of domain
[   11.933305] systemd[193]: Failed to add rule for system call spu_run() / -10189, ignoring: Numerical argument out of domain
[   11.933637] systemd[193]: Failed to add rule for system call kexec_file_load() / 320, ignoring: Numerical argument out of domain
[   11.933732] systemd[193]: Failed to add rule for system call switch_endian() / -10191, ignoring: Numerical argument out of domain
[   11.934297] systemd[193]: Failed to add rule for system call break() / -10003, ignoring: Numerical argument out of domain
[   11.934420] systemd[193]: Failed to add rule for system call uselib() / 134, ignoring: Numerical argument out of domain
[   11.934514] systemd[193]: Failed to add rule for system call profil() / -10040, ignoring: Numerical argument out of domain
[   11.934606] systemd[193]: Failed to add rule for system call vm86() / -10071, ignoring: Numerical argument out of domain
[   11.934742] systemd[193]: Failed to add rule for system call mpx() / -10030, ignoring: Numerical argument out of domain
[   11.934882] systemd[193]: Failed to add rule for system call s390_runtime_instr() / -10196, ignoring: Numerical argument out of domain
[   11.934975] systemd[193]: Failed to add rule for system call vm86old() / -10072, ignoring: Numerical argument out of domain
[   11.935087] systemd[193]: Failed to add rule for system call rtas() / -10187, ignoring: Numerical argument out of domain
[   11.935186] systemd[193]: Failed to add rule for system call lock() / -10027, ignoring: Numerical argument out of domain
[   11.935284] systemd[193]: Failed to add rule for system call prof() / -10039, ignoring: Numerical argument out of domain
[   11.935362] systemd[193]: Failed to add rule for system call _sysctl() / 156, ignoring: Numerical argument out of domain
[   11.935457] systemd[193]: Failed to add rule for system call query_module() / 178, ignoring: Numerical argument out of domain
[   11.935544] systemd[193]: Failed to add rule for system call gtty() / -10022, ignoring: Numerical argument out of domain
[   11.935668] systemd[193]: Failed to add rule for system call umount() / -10070, ignoring: Numerical argument out of domain
[   11.935772] systemd[193]: Failed to add rule for system call vserver() / 236, ignoring: Numerical argument out of domain
[   11.935849] systemd[193]: Failed to add rule for system call idle() / -10023, ignoring: Numerical argument out of domain
[   11.935953] systemd[193]: Failed to add rule for system call stty() / -10065, ignoring: Numerical argument out of domain
[   11.936058] systemd[193]: Failed to add rule for system call create_module() / 174, ignoring: Numerical argument out of domain
[   11.936158] systemd[193]: Failed to add rule for system call s390_pci_mmio_write() / -10198, ignoring: Numerical argument out of domain
[   11.936230] systemd[193]: Failed to add rule for system call ftime() / -10013, ignoring: Numerical argument out of domain
[   11.936309] systemd[193]: Failed to add rule for system call pciconfig_write() / -10088, ignoring: Numerical argument out of domain
[   11.936398] systemd[193]: Failed to add rule for system call ulimit() / -10069, ignoring: Numerical argument out of domain
[   11.936553] systemd[193]: Failed to add rule for system call bdflush() / -10002, ignoring: Numerical argument out of domain
[   11.936640] systemd[193]: Failed to add rule for system call get_kernel_syms() / 177, ignoring: Numerical argument out of domain
[   11.936816] systemd[193]: Failed to add rule for system call stime() / -10064, ignoring: Numerical argument out of domain
[   11.936897] systemd[193]: Failed to add rule for system call pciconfig_read() / -10087, ignoring: Numerical argument out of domain
[   11.936979] systemd[193]: Failed to add rule for system call s390_pci_mmio_read() / -10197, ignoring: Numerical argument out of domain
[   11.937064] systemd[193]: Failed to add rule for system call pciconfig_iobase() / -10086, ignoring: Numerical argument out of domain
[   11.937239] systemd[193]: Failed to add rule for system call spu_run() / -10189, ignoring: Numerical argument out of domain
[   11.937329] systemd[193]: Failed to add rule for system call ssetmask() / -10061, ignoring: Numerical argument out of domain
[   11.937472] systemd[193]: Failed to add rule for system call sgetmask() / -10053, ignoring: Numerical argument out of domain
[   11.937574] systemd[193]: Failed to add rule for system call switch_endian() / -10191, ignoring: Numerical argument out of domain
[   11.938026] systemd[193]: Failed to add rule for system call break() / -10003, ignoring: Numerical argument out of domain
[   12.132673] systemd[193]: Failed to add rule for system call profil() / -10040, ignoring: Numerical argument out of domain
[   12.202554] systemd[193]: Failed to add rule for system call vm86() / -10071, ignoring: Numerical argument out of domain
[   12.322444] systemd[193]: Failed to add rule for system call mpx() / -10030, ignoring: Numerical argument out of domain
[   12.475129] systemd[193]: Failed to add rule for system call s390_runtime_instr() / -10196, ignoring: Numerical argument out of domain
[   12.592965] systemd[193]: Failed to add rule for system call vm86old() / -10072, ignoring: Numerical argument out of domain
[   12.799209] systemd[193]: Failed to add rule for system call rtas() / -10187, ignoring: Numerical argument out of domain
[   12.882251] systemd[193]: Failed to add rule for system call lock() / -10027, ignoring: Numerical argument out of domain
[   13.131330] systemd[193]: Failed to add rule for system call prof() / -10039, ignoring: Numerical argument out of domain
[   13.214367] systemd[193]: Failed to add rule for system call gtty() / -10022, ignoring: Numerical argument out of domain
[   13.383630] systemd[193]: Failed to add rule for system call umount() / -10070, ignoring: Numerical argument out of domain
[   13.679228] systemd[193]: Failed to add rule for system call idle() / -10023, ignoring: Numerical argument out of domain
[   14.500489] systemd[193]: Failed to add rule for system call stty() / -10065, ignoring: Numerical argument out of domain
[   14.537037] systemd[193]: Failed to add rule for system call s390_pci_mmio_write() / -10198, ignoring: Numerical argument out of domain
[   14.591834] systemd[193]: Failed to add rule for system call ftime() / -10013, ignoring: Numerical argument out of domain
[   14.665972] systemd[193]: Failed to add rule for system call pciconfig_write() / -10088, ignoring: Numerical argument out of domain
[   14.721738] systemd[193]: Failed to add rule for system call ulimit() / -10069, ignoring: Numerical argument out of domain
[   14.721828] systemd[193]: Failed to add rule for system call bdflush() / -10002, ignoring: Numerical argument out of domain
[   14.797441] systemd[193]: Failed to add rule for system call stime() / -10064, ignoring: Numerical argument out of domain
[   14.835581] systemd[193]: Failed to add rule for system call pciconfig_read() / -10087, ignoring: Numerical argument out of domain
[   14.835651] systemd[193]: Failed to add rule for system call s390_pci_mmio_read() / -10197, ignoring: Numerical argument out of domain
[   14.835715] systemd[193]: Failed to add rule for system call pciconfig_iobase() / -10086, ignoring: Numerical argument out of domain
[   14.835792] systemd[193]: Failed to add rule for system call spu_run() / -10189, ignoring: Numerical argument out of domain
[   14.835859] systemd[193]: Failed to add rule for system call ssetmask() / -10061, ignoring: Numerical argument out of domain
[   15.030547] systemd[193]: Failed to add rule for system call sgetmask() / -10053, ignoring: Numerical argument out of domain
[   15.030621] systemd[193]: Failed to add rule for system call switch_endian() / -10191, ignoring: Numerical argument out of domain
```

### Used distribution

> Linux arch 4.16.2-2-ARCH #1 SMP PREEMPT Tue Apr 17 19:37:34 UTC 2018 x86_64 GNU/Linux

### In case of bug report: Expected behaviour you didn't see

> Data errors should not be in dmesg

### In case of bug report: Unexpected behaviour you saw

> A large number of errors!

### In case of bug report: Steps to reproduce the problem

> Set the arch archive to the latest time and run the pacman -S systemd command, you will have systemd-238.76-1 which will have the error data in dmesg.

**Comment by poettering** [comment]
This is logged at debug level and hence not visible unless you explicitly ask to see it by enabling debug logging. 

The fact that you see this is not an issue at all. We define our seccomp profiles comprehensively, i.e. listing all syscalls relevant for any of the archs. Since various syscalls are specific to some archs it's completely expect that our attempts to block some of them doesn't work on some archs. I figure you run an x86-64 system, which means we'll apply syscall filtering to three sets of syscall APIs: x86-64, i386 and x32. i386 is really old and carries a ton of cruft since removed in x86-64, which means attempts to block them on x86-64 will result in EDOM errors, but again that's entirely fine and expected.

Long story short, all is good, nothing to see here, this is only interesting for developers who want to check that all is good, and by turning on debug logging you asked for this kind of information and got it.

Hope that makes sense. Closing.
